pkg/apis/batch/validation/validation.go

```diff
@@ -425,6 +548,26 @@ func validateJobStatus(status *batch.JobStatus, fldPath *field.Path) field.Error
 			}
 		}
 	}
+	if opts.RejectSuccessCriteriaMetForNonIndexedJob {
+		if IsConditionTrue(job.Status.Conditions, batch.JobSuccessCriteriaMet) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("conditions"), field.OmitValueType{}, "cannot set SuccessCriteriaMet to NonIndexed Job"))
+		}
+	}
+	if opts.RejectSuccessCriteriaMetWithFailedCondition {
+		if IsConditionTrue(job.Status.Conditions, batch.JobSuccessCriteriaMet) && IsJobFailed(job) {
```

Comment:
create an `IsJobSuccessCriteriaMet` peer to `IsJobComplete`

Reply:
That makes sense.

Reply:
Done.